### PR TITLE
First version of pLDDT and LUR module

### PR DESCRIPTION
**Pull request by bordin89**
First version of pLDDT and LUR module.
Input: CIF File
Output: Summary with Average pLDDT (by chain or domain depending on input) and percentage of residues in Long Unordered Regions (LUR, by chain or domain depending on input).
Add additional models, constant (for min_LUR_length) and writer.
Requires biopython for MMCIF reader.

### CODE REVIEW

File: cath_alphaflow/commands/extract_plddt_and_lur.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    if acc_id is None:
+        acc_id = cif_path.stem
+    mmcif_dict = MMCIF2Dict.MMCIF2Dict(cif_path)
+    chain_plddt = mmcif_dict["_ma_qa_metric_global.metric_value"][0]
+    plddt_string = mmcif_dict["_ma_qa_metric_local.metric_value"]
+    segment_plddt = ""
+    if chopping:
+        for segment in chopping.segments:
+            segment_plddt += plddt_string[(segment.start - 1) : segment.end]
+            domain_length = len(segment_plddt)
```
Comment (iandaaci) [inline cath_alphaflow/commands/extract_plddt_and_lur.py:79]: `domain_length` needs to be a running total of the segments, then take the average after the loop, e.g.

```
domain_length = 0
for segment in chopping.segments:
            segment_plddt += plddt_string[(segment.start - 1) : segment.end]
            domain_length += len(segment_plddt)
average_plddt = ...
```

File: cath_alphaflow/commands/extract_plddt_and_lur.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            average_plddt = round((sum(segment_plddt) / domain_length) * 100, 2)
```
Comment (iandaaci) [inline cath_alphaflow/commands/extract_plddt_and_lur.py]: not sure about the logic of this average calculation - is segment_plddt an `int` or a list of `int`? (`sum` usually takes an `iterable`)

File: cath_alphaflow/commands/extract_plddt_and_lur.py (hunk 1 above)
Comment (bordin89) [reply cath_alphaflow/commands/extract_plddt_and_lur.py]: Fixed in 'move domain_length out of loop to solve average'

File: cath_alphaflow/commands/extract_plddt_and_lur.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+def get_average_plddt_from_plddt_string(
+    cif_path: Path, *, chopping=None, acc_id=None
+) -> pLDDTSummary:
+    if acc_id is None:
+        acc_id = cif_path.stem
+    mmcif_dict = MMCIF2Dict.MMCIF2Dict(cif_path)
+    chain_plddt = mmcif_dict["_ma_qa_metric_global.metric_value"][0]
+    plddt_string = mmcif_dict["_ma_qa_metric_local.metric_value"]
+    segment_plddt = ""
```
Comment (sillitoe) [inline cath_alphaflow/commands/extract_plddt_and_lur.py:75]: should this be an array of `int` that you append to in the following loop?

```
segment_plddt = []
```

**Pull request merged**